### PR TITLE
Project page improvements

### LINKED ISSUE #454: Active projects filter
It would be good to have a sub-page of [projects](https://rse.shef.ac.uk/service/projects/), e.g. `https://rse.shef.ac.uk/service/projects/active` which had the list with completed projects hidden.

Not sure how feasible this would be via Jekyll, but it could be useful in cases where we want to talk about what we're actively doing. e.g. @bobturneruk's application for a summer intern, who might help out on one or more of our active projects.

### PR DESCRIPTION
**Pull request by Robadob**
Started working on #454, ended up doing alot more than planned.

**TLDR:** Clicking a tag from the tag cloud, or one of the new filter links, adds `#<tag>` to the address, bolds the clicked item, and hides unrelated items from the project list. If the page loads with the same `#<tag>` in `window.location.href` it simulates the user clicking the associated link. Also fix a couple of minor issues with the tag cloud (`&nbsp` and FLAMEGPU dupe).

Current progress visible [on my mirror site](https://rse-mirror.robadob.org/service/projects)

- [x] ~~Add hidden [active projects page](https://rse-mirror.robadob.org/service/projects/active)~~ (This is redundant)
- [x] Make the projects tag cloud clickable to filter projects
- [x] Replace spaces with `&nbsp` inside tag cloud tags. (to fix "static analysis" wrapping)
- [x] Unify `FLAMEGPU` and `FLAME GPU` tags
- [x] Add filter links for active/completed projects
- [x] On page load, auto parse `#tag` from url, to pretend link was clicked
- [ ] Can the delay on making the clicked tag bold be improved?? (or hidden as a lazy hack with `:hover`)

*Potential issues:*
* Encoding tags to valid css classes via liquid is awkward. I currently escape `.&#+` (and spaces), but there are probably others that will break stuff. I did try the spec documented use of `\` but that didn't seem to work with jquery selectors.
* If `active` or `completed` are added as technical tags, they will clash with the filter links, causing some undefined behaviour.

Closes #454

### CODE REVIEW

**Comment by Robadob** [comment]
Can you give this a review @ptheywood and let me know if I've done anything awful, and whether you think it's worth addressing my final item from OP.

[Mirror site](https://rse-mirror.robadob.org/service/projects) has all commits (except reverting the hidden page, and also has items from #456).

Thanks

File: assets/js/main.js [hunk 1]
```diff
@@ -113,5 +113,29 @@ jQuery(document).ready(function($) {
         }
 
     });
+    
+    /* ======= Projects Tag Cloud ======= */
+    $("a.tag-link").on( "click", function(e) {
+        var tag = $(this).attr("href").substring($(this).attr("href").lastIndexOf('#') + 1);
```
Comment (ptheywood) [inline assets/js/main.js]: Could use `window.location.hash` instead, although it includes the `#` char in the value, but this works so no need to change it.

File: pages/service/projects/projects.md [hunk 2]
```diff
@@ -71,4 +75,14 @@ Some projects we have worked on (not a comprehensive list):
         <hr/>
     {% endfor %}
 </div>
-
+<script>
+window.addEventListener('load', (event) => {
+    var offset = window.location.href.lastIndexOf('#');
```
Comment (ptheywood) [inline pages/service/projects/projects.md]: Same option of using `window.location.hash` with `.slice(1)`, but agian not neccesary just slightly cleaner code.

**Review by ptheywood — APPROVED** [review]

File: assets/js/main.js (hunk 1 above)
Comment (Robadob) [reply assets/js/main.js]: Not sure `window.location.hash` would be appropriate here, because it's trying to strip the `#tag` part from the href of an anchor.
